cfgmgr/coppmgr.cpp

```diff
@@ -0,0 +1,859 @@
+                        if ((m_coppGroupRestrictedMap.find(trap_group) 
+                             != m_coppGroupRestrictedMap.end()) &&
+                             (checkIfTrapGroupFeaturePending(trap_group)))
+                        {
+                            vector<FieldValueTuple> fvs;
+                            vector<FieldValueTuple> modified_fvs;
+                            string trap_ids;
+
+                            for (auto i : m_coppGroupRestrictedMap[trap_group])
+                            {
+                                FieldValueTuple fv(i.first, i.second);
+                                fvs.push_back(fv);
+                            }
+                            getTrapGroupTrapIds(trap_group, trap_ids);
+                            if (!trap_ids.empty())
+                            {
+                                FieldValueTuple fv(COPP_TRAP_ID_LIST_FIELD, trap_ids);
+                                fvs.push_back(fv);
+                            }
+                            coppGroupGetModifiedFvs (trap_group, fvs, modified_fvs, false);
+                            if (!modified_fvs.empty())
+                            {
+                                m_appCoppTable.set(trap_group, modified_fvs);
+                            }
+                            setCoppGroupStateOk(trap_group);
+                        }
+                    }
+                    m_coppTrapDisabledMap.erase(it.first);
+                }
+            }
+            else
+            {
+                if (m_coppTrapDisabledMap.find(it.first) == m_coppTrapDisabledMap.end())
+                {
+                    m_coppTrapDisabledMap[it.first] = true;
+                    if (m_coppTrapIdTrapGroupMap.find(it.first) != m_coppTrapIdTrapGroupMap.end())
+                    {
+                        string trap_group = m_coppTrapIdTrapGroupMap[it.first];
+
+                        /* Delete the restricted copp group when it goes to pending state
+                         * on feature disable
+                         */
+                        if ((m_coppGroupRestrictedMap.find(trap_group) 
+                             != m_coppGroupRestrictedMap.end()) &&
+                             (checkIfTrapGroupFeaturePending(trap_group)))
+                        {
+                            m_appCoppTable.del(trap_group);
+                            delCoppGroupStateOk(trap_group);
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+bool CoppMgr::isTrapDisabled(string trap_id)
+{
+    auto trap_idx = m_coppTrapDisabledMap.find(trap_id);
+    if (trap_idx != m_coppTrapDisabledMap.end())
+    {
+        return m_coppTrapDisabledMap[trap_id];
+    }
+    return false;
+}
+
+/* The genetlink  fields are restricted and needs to be installed only on
+ * feature(sflow) enable
+ */
+bool CoppMgr::coppGroupHasRestrictedFields (vector<FieldValueTuple> &fvs)
```

Comment:
I dont think we should have a check for Restricted Fields. Lets discuss offline

Reply:
Removed this logic